File: 2017/APR/LTIME47/BEARSEG/BEARSEG.cpp
// Code written by Monal
#include "bits/stdc++.h"
 
using namespace std;
 
#define ll long long int
#define l long int
 
 
void sum_of_subsequence(l n,ll P) {
    ll curr_sum;
    vector<ll> arr(n,0);
    for (l i = 0; i < n; ++i) {
    	cin>>arr[i];
    }
    ll max=0,occ=0;
    for (l i = 0; i < n; ++i) {
        curr_sum = 0;
        for (l j = i; j < n; ++j) {
           curr_sum = curr_sum + arr[j];
           if (curr_sum % P > max) {
            	max = curr_sum % P;
            	occ=1;
           }
           else if (curr_sum % P == max) {
           	occ++;
           }
        }
    }
    arr.clear();
    cout<<max<<" "<<occ<<"\n";
}
 
int main() {
	ios::sync_with_stdio(false);
	int tc;
	cin>>tc;
	while (tc--) {
		l n;
		ll p;
		cin>>n>>p;
		sum_of_subsequence(n,p);
	}
	return 0;
}

File: 2017/APR/LTIME47/BEARSEG/BEARSEG.py
# Code written by Monal
def sum_of_subsequences(n, p):
    arr = list(map(int, input().split()))
    maxi = 0
    occ = 0
    for i in range(n):
        curr_sum = 0
        for j in range(i, n):
            curr_sum = curr_sum + arr[j]
            if curr_sum % p > maxi:
                maxi = curr_sum % p
                occ = 1
            elif curr_sum % p == maxi:
                occ += 1
    print(maxi, occ)


def main():
    tc = int(input())
    for _ in range(tc):
        n, p = map(int, input().split())
        sum_of_subsequences(n, p)


if __name__ == '__main__':
    main()
